base_layer/core/src/covenants/output_set.rs

```diff
@@ -80,6 +82,9 @@ impl<'a> OutputSet<'a> {
         self.0.symmetric_difference(&other.0).copied().collect()
     }
 
+    /// Given a predicate on `TransactionOutput`, it finds a present instance, in place.
+    /// If no `TransactionOutput` is found, it clears the underlying set, in place. Otherwise,
+    /// it keeps this sole `TransactionOutput` instance.
```

Comment:
Finds a single matching output, modifying the output set in place. If no matching output is found, the output set will be empty.